drizzlepac/haputils/quality_analysis.py

```diff
@@ -425,9 +446,257 @@ def generate_plots(json_data):
+    tips : list
+        List of indices for the HOVER_COLUMNS entries to be used as tooltips 
+        to be included in the hover tool.
+
+    """
+    tools = [(TOOLTIPS_LIST[i], '@{}{}{}'.format(
+                                TOOLSEP_START, 
+                                HOVER_COLUMNS[i],
+                                TOOLSEP_END)) for i in tips]
+    
+    return tools
+    
+def get_pandas_data(csv_filename):
+    """Load the harvested data, stored in a CSV file, into local arrays.
+
+    Parameters
+    ==========
+    csv_filename: str
+    Name of the CSV file created by the harvester.
+
+    Returns
+    =======
+    phot_data: Pandas dataframe
+    Dataframe which is a subset of the input Pandas dataframe written out as
+    a CSV file.  The subset dataframe consists of only the requested columns
+    and rows where all of the requested columns did not contain NaNs.
+
+    (mean_dMagAp1_mean, mean_dMagAp1_median): tuple
+    Aperture 1 mean of means and mean of medians
+
+    (mean_dMagAp2_mean, mean_dMagAp2_median): tuple
+    Aperture 2 mean of means and mean of medians
+    """
+    
+    # Instantiate a Pandas Dataframe Reader (lazy instantiation)
+    # df_handle = PandasDFReader_CSV("svm_qa_dataframe.csv")
+    df_handle = PandasDFReader(csv_filename, log_level=logutil.logging.NOTSET)
+
+    # In this particular case, the names of the desired columns do not
+    # have to be further manipulated, for example, to add dataset specific
+    # names.
+    # 
+    # Get the relevant column data, eliminating all rows which have NaNs
+    # in any of the relevant columns.
+    if csv_filename.endswith('.h5'):
+        fit_data = df_handle.get_columns_HDF5(HOVER_COLUMNS + RESULTS_COLUMNS)
+        source_data = df_handle.get_columns_HDF5(HOVER_COLUMNS + SOURCE_COLUMNS)
+    else:
+        fit_data = df_handle.get_columns_CSV(HOVER_COLUMNS + RESULTS_COLUMNS)
+        source_data = df_handle.get_columns_CSV(HOVER_COLUMNS + SOURCE_COLUMNS)
+
+    return fit_data, source_data
+    
+def build_circle_plot(**plot_dict):
+    """Create figure object for plotting desired columns as a scatter plot with circles"""
+    # Interpret required elements
+    x = plot_dict['x']
+    y = plot_dict['y']
+    title = plot_dict['title']
+    x_label = plot_dict['x_label']
+    y_label = plot_dict['y_label']
+    tips = plot_dict['tips']
+    source = plot_dict['source']
+    
+    # check for optional elements
+    color = plot_dict.get('color', 'blue')
+    click_policy = plot_dict.get('click_policy', 'hide')
+    markersize = plot_dict.get('markersize', 10)
+    colormap = plot_dict.get('colormap', False)
+    legend_group = plot_dict.get('legend_group')
+    
+    # Define a figure object
+    p1 = figure()
+
+    if colormap:
+        # Add the glyphs
+        p1.circle(x=x, y=y, source=source, 
+                  fill_alpha=0.5, line_alpha=0.5, 
+                  size=markersize, color='colormap',
+                  legend_group=legend_group)
```

Comment:
Curious - what is 'colormap' here?

Reply:
A comment was added to clarify the use of 'colormap' here.  In addition, the function docstring was updated to include a full description of all parameters and return values which explains the use of this column name for the colors used in the plot. All this was included in 7ba7497.